monitor/internal/pod/controller.go

```diff
@@ -116,38 +128,38 @@ type ReconcilePod struct {
 // Reconcile reads that state of the cluster for a pod object
 func (r *ReconcilePod) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	ctx := context.Background()
-	puID := request.NamespacedName.String()
+	nn := request.NamespacedName.String()
 
 	// Fetch the corresponding pod object.
 	pod := &corev1.Pod{}
 	if err := r.client.Get(ctx, request.NamespacedName, pod); err != nil {
 		if errors.IsNotFound(err) {
-			zap.L().Debug("Pod IsNotFound", zap.String("puID", puID))
-			handlePUCtx, handlePUCancel := context.WithTimeout(ctx, r.handlePUEventTimeout)
-			defer handlePUCancel()
-
-			// call destroy - the policy engine is going to call stop first for us if necessary
-			if err := r.handler.Policy.HandlePUEvent(
-				handlePUCtx,
-				puID,
-				common.EventDestroy,
-				policy.NewPURuntimeWithDefaults(),
-			); err != nil {
-				zap.L().Error("failed to handle destroy event", zap.String("puID", puID), zap.Error(err))
-			}
+			r.deleteReconcileCh <- struct{}{}
```

Comment:
Just curious,I see you send a event to delete the PU,  I see no harm  in the below logic to be removed. Was there any specific reason?